### PR TITLE
Clean models

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Clean models - use DataMapper instead of Active Record pattern.
I duplicate code here which could be solved by crating a custom Repository. I would recommend to not create a custom repository but change the query from a publicKey find to an EMail find in the future.

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Review by einhornimmond — APPROVED** [review]
no comment

**Review by Mogge — APPROVED** [review]
Cool!

File: backend/src/graphql/resolvers/GdtResolver.ts [hunk 1]
```diff
@@ -23,7 +23,10 @@ export class GdtResolver {
     if (!result.success) throw new Error(result.data)
 
     // load user
-    const userEntity = await dbUser.findByPubkeyHex(result.data.user.public_hex)
+    const userEntity = await dbUser
+      .createQueryBuilder('user')
+      .where('hex(user.pubkey) = :pubkeyHex', { pubkeyHex: result.data.user.public_hex })
+      .getOneOrFail()
```
Comment (Elweyn) [inline backend/src/graphql/resolvers/GdtResolver.ts:29]: Same as BalanceResolver userEntity

**Review by Elweyn — CHANGES_REQUESTED** [review]

**Comment by ulfgebhardt** [comment]
Closing in favor of #927 and #928

**Pull request closed without merging**